apps/lwipserver2/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+
+cmake_minimum_required(VERSION 3.8.2)
+
+project(lwipserver_single C)
+includeGlobalComponents()
+
+find_package(projects_libs REQUIRED)
+
+set(CAmkESCPP ON CACHE BOOL "" FORCE)
+if("${KernelArch}" STREQUAL "x86")
+    set(cpp_define -DKernelArchX86)
+elseif("${KernelArch}" STREQUAL "arm")
+    set(cpp_define -DKernelArchArm)
+endif()
```

Comment:
let's give it a clear error message here for anything else and avoid potentially undefined default behavior
```
else()
    message(FATAL_ERROR "Unsupported architecture.")
endif()
```